rpc/coreclient.go

```diff
@@ -0,0 +1,71 @@
+package rpc
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/celestiaorg/celestia-core/rpc/client/http"
+	ctypes "github.com/celestiaorg/celestia-core/rpc/core/types"
+	"github.com/celestiaorg/celestia-core/types"
+)
+
+const newBlockSubscriber = "NewBlock/Events"
+
+// Client represents an RPC client designed to communicate
+// with Celestia Core.
+type Client struct {
+	http       *http.HTTP
```

Comment:
If you and Hlib agree on the local / in process client, then it kinda makes sense that this field is an interface instead: https://github.com/tendermint/tendermint/blob/master/rpc/client/interface.go

But I think this is not too important to change right now.